ionoscloud/data_source_dbaas_mariadb_cluster.go

```diff
@@ -0,0 +1,160 @@
+				Computed:    true,
+			},
+			"storage_size": {
+				Type:        schema.TypeInt,
+				Description: "The amount of storage per instance in gigabytes (GB).",
+				Computed:    true,
+			},
+			// TODO -- Check if we need to include the credentials in the DS based on what we receive
+			// TODO -- as response, maybe we need to include only the username.
+			"connections": {
+				Type:        schema.TypeList,
+				Description: "The network connection for your cluster. Only one connection is allowed.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"datacenter_id": {
+							Type:        schema.TypeString,
+							Description: "The datacenter to connect your cluster to.",
+							Computed:    true,
+						},
+						"lan_id": {
+							Type:        schema.TypeString,
+							Description: "The numeric LAN ID to connect your cluster to.",
+							Computed:    true,
+						},
+						"cidr": {
+							Type:        schema.TypeString,
+							Description: "The IP and subnet for your cluster.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				Description: "A weekly 4 hour-long window, during which maintenance might occur.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:        schema.TypeString,
+							Description: "Start of the maintenance window in UTC time.",
+							Computed:    true,
+						},
+						"day_of_the_week": {
+							Type:        schema.TypeString,
+							Description: "The name of the week day.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name pointing to your cluster.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceMariaDBClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+	id, idOk := d.GetOk("id")
+	name, nameOk := d.GetOk("display_name")
+
+	if idOk && nameOk {
+		return diag.FromErr(fmt.Errorf("ID and display_name cannot be both specified at the same time"))
+	}
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("please provide either the MariaDB cluster ID or display_name"))
+	}
+
+	var cluster dbaas.ClusterResponse
+	var err error
+
+	if idOk {
+		/* search by ID */
+		cluster, _, err = client.GetCluster(ctx, id.(string))
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching the MariaDB cluster with ID %v: %w", id.(string), err))
+		}
+	} else {
+		clusters, _, err := client.ListClusters(ctx, "")
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching MariaDB clusters: %w", err))
+		}
+
+		var results []dbaas.ClusterResponse
+
+		if clusters.Items != nil && len(*clusters.Items) > 0 {
```

Comment:
can shorten to just `if clusters.Items != nil` since there is only the for loop under this condition

Reply:
@cbeti-ionos done in eabe0a8e1529e0c3b0c3d8b882cc4b2d6056674f